File: frontend/mobile/src/api/services/auth/oauth2Service.ts
import { authApi } from '@/src/api';
import type { paths } from '@/src/api/generated/openapi';

type GoogleLoginRequest =
  paths['/api/v1/oauth2/google']['post']['requestBody']['content']['application/json'];

type LoginResponse =
  paths['/api/v1/oauth2/google']['post']['responses'][200]['content']['*/*'];

export const googleLogin = async (token: string) => {
  const body: GoogleLoginRequest = {
    token
  };

  const { data } = await authApi.post<LoginResponse>(
    '/api/v1/oauth2/google',
    body
  );

  return data;
};

File: frontend/phone/lib/features/auth/services/oauth2_service.dart
import 'package:phone/generated/models/login_response.dart';
import 'package:phone/generated/models/o_auth2_login_request.dart';
import 'package:phone/core/api/api.dart';

class OAuth2Service {
  Future<LoginResponse> googleLogin(String token) async {
    final requestBody = OAuth2LoginRequest(token: token);

    final response = await api.post<Map<String, dynamic>>(
      '/api/v1/oauth2/google',
      data: requestBody.toJson(),
    );

    if (response.data == null) {
      throw Exception('Response data was null');
    }

    return LoginResponse.fromJson(response.data!);
  }
}

final OAuth2Service oAuth2Service = OAuth2Service();
